wordpress-mobile/WordPress-iOS
# Initial themes browser implementation

Closes #3902; closes #4176 

Updates theme list endpoints to v1.2 as recommended by Hyperion team and implements basic theme browser following design [here](https://teamhyperion.wordpress.com/2015/05/07/calypso-theme-showcase-in-calypso-basic-master-thread/). Note that searching, filtering, and sundry theme actions are covered by other [tickets](https://github.com/wordpress-mobile/WordPress-iOS/labels/Themes).

To test: 
1. Enable themes feature with `wpdebug://themes?enabled=1` as discussed in #3913
2. Activate themes browser from the blog details screen
3. Watch available themes synchronize with server
4. Check that browser adapts layout sensibly to rotation and assorted device sizes
5. Browse theme demo site by selecting its cell.

@jleandroperez Could you review please?
Needs Review: @jleandroperez
